app/lib/actions/katello/repository/create.rb

```diff
@@ -50,7 +53,10 @@ def run
           ::User.current = ::User.anonymous_api_admin
           unless input[:clone]
             repository = ::Katello::Repository.find(input[:repository_id])
-            ForemanTasks.async_task(Katello::Repository::MetadataGenerate, repository)
+
+            # Container push repos do not need metadata generation
+            ForemanTasks.async_task(Katello::Repository::MetadataGenerate, repository) unless
+              repository.root.is_container_push
```

Comment:
Rather than skipping just the async_task call for MetadataGenerate, you could actually skip this entire section up on line 38: 
```
            concurrence do
              plan_self(:repository_id => repository.id, :clone => clone)
              if !clone && repository.url.present?
                repository.product.alternate_content_sources.with_type(repository.content_type).each do |acs|
                  acs.smart_proxies.each do |smart_proxy|
                    smart_proxy_acs = ::Katello::SmartProxyAlternateContentSource.create(alternate_content_source_id: acs.id, smart_proxy_id: smart_proxy.id, repository_id: repository.id)
                    plan_action(Pulp3::Orchestration::AlternateContentSource::Create, smart_proxy_acs)
                  end
                end
              end
```

This section runs `plan_self`, which plans the run phase of the action (what's under `def run`).  The run phase only triggers the MetadataGenerate call. By skipping the entire run phase, overhead from creating a job for a Dynflow worker will be completely removed.

You migh task if the other stuff relating to alternate content sources is important. Alternate content sources only help with syncing, and push repositories will never sync anything, so there should be no reason to even create an ACS for them. We'll save some more cycles by skipping that as well.

I don't see anything else in this action that can be skipped for container push repos.

Reply:
Thank you for finding this :) 
Your suggested changes have been implemented.